src/client/src/app/lib/logging/logger.ts

```diff
@@ -0,0 +1,25 @@
+type LogLevel = 'info' | 'warn' | 'error';
+
+export function logEvent(
+  level: LogLevel,
+  event: string,
+  payload: Record<string, any>,
```

Comment:
You've probably also noticed this generates a warning because of the use of `any` which is a smell in typescript. Looks like in every use case of this func we're passing it an object, so we have two options here: 

- You could just change `any` to `unknown`, which tells Typescript we are explicitly declaring this as an unknown type.
- Or to do this a more proper Typescript way, we could label this func with a generic type, along the lines of: 
<img width="184" height="92" alt="image" src="https://github.com/user-attachments/assets/46d94b0f-a7f7-49bb-9058-b261aa1b1172" />

This should work with all the current places you're using this